src/object_store/src/object/s3.rs

```diff
@@ -724,7 +724,7 @@ impl S3ObjectStore {
                 // If both conditions are met, it is considered that there is a risk of data deletion.
                 match rule.filter().as_ref() {
                     Some(&LifecycleRuleFilter::Prefix(prefix)) => {
-                        if let Some(ExpirationStatus::Enabled) = rule.status && prefix == data_directory{
+                        if let Some(ExpirationStatus::Enabled) = rule.status && data_directory.contains(prefix){
```

Comment:
hmm... Shouldn't this be `prefix.starts_with(data_directory)`?

Reply:
I think it should be `data_directory.start_with(prefix)`, eg, `prefix=hummock`, `data_directory = hummock_001`, and hummock_001 has rule.

Reply:
I get it wrong, it should be `data_directory.start_with(prefix)`.